cmd/lakectl/cmd/fs.go

```diff
@@ -409,6 +413,159 @@ var fsRmCmd = &cobra.Command{
 	},
 }
 
+const (
+	fsDownloadCmdMinArgs = 1
+	fsDownloadCmdMaxArgs = 2
+
+	fsDownloadParallelDefault = 6
+)
+
+var fsDownloadCmd = &cobra.Command{
+	Use:   "download <path uri> [<destination path>]",
+	Short: "Download object(s) from a given repository path",
+	Args:  cobra.RangeArgs(fsDownloadCmdMinArgs, fsDownloadCmdMaxArgs),
+	Run: func(cmd *cobra.Command, args []string) {
+		pathURI := MustParsePathURI("path", args[0])
+		flagSet := cmd.Flags()
+		direct := MustBool(flagSet.GetBool("direct"))
+		recursive := MustBool(flagSet.GetBool("recursive"))
+		parallel := MustInt(flagSet.GetInt("parallel"))
+
+		if parallel < 1 {
+			DieFmt("Invalid value for parallel (%d), minimum is 1.\n", parallel)
+		}
+
+		// optional destination directory
+		var dest string
+		if len(args) > 1 {
+			dest = args[1]
+			// verify destination path exists
```

Comment:
```suggestion
			// verify destination path is valid
```
The comment is misleading as this will only fail for invalid `dest` path - the result from `pathExists` is ignored, plus the parent paths are created later, as ppart of the download.